test/js-api/exception/identity.tentative.any.js

```diff
@@ -3,59 +3,68 @@
 // META: script=/wasm/jsapi/wasm-module-builder.js
 
 test(() => {
-  const tag = new WebAssembly.Tag({ parameters: ["i32"] });
-  const exn = new WebAssembly.Exception(tag, [42]);
-  const exn_same_payload = new WebAssembly.Exception(tag, [42]);
-  const exn_diff_payload = new WebAssembly.Exception(tag, [53]);
-
   const builder = new WasmModuleBuilder();
-  const jsfuncIndex = builder.addImport("module", "jsfunc", kSig_v_v);
-  const tagIndex = builder.addImportedTag("module", "tag", kSig_v_i);
+
+  // Tag defined in JavaScript and imported into Wasm
+  const jsTag = new WebAssembly.Tag({ parameters: ["i32"] });
+  const jsTagIndex = builder.addImportedTag("module", "jsTag", kSig_v_i)
+  const jsTagExn = new WebAssembly.Exception(jsTag, [42]);
+  const jsTagExnSamePayload = new WebAssembly.Exception(jsTag, [42]);
+  const jsTagExnDiffPayload = new WebAssembly.Exception(jsTag, [53]);
+  const throwJSTagExnIndex = builder.addImport("module", "throwJSTagExn", kSig_v_v);
+
   const imports = {
     module: {
-      jsfunc: function() { throw exn; },
-      tag: tag
+      throwJSTagExn: function() { throw jsTagExn; },
+      jsTag: jsTag
     }
   };
 
   builder
-    .addFunction("catch_rethrow", kSig_v_v)
+    .addFunction("catch_js_tag_rethrow", kSig_v_v)
     .addBody([
       kExprTry, kWasmStmt,
-        kExprCallFunction, jsfuncIndex,
-      kExprCatch, tagIndex,
+        kExprCallFunction, throwJSTagExnIndex,
+      kExprCatch, jsTagIndex,
         kExprDrop,
         kExprRethrow, 0x00,
       kExprEnd
     ])
     .exportFunc();
 
   builder
-    .addFunction("catch_all_rethrow", kSig_v_v)
+    .addFunction("catch_all_js_tag_rethrow", kSig_v_v)
     .addBody([
       kExprTry, kWasmStmt,
-        kExprCallFunction, jsfuncIndex,
+        kExprCallFunction, throwJSTagExnIndex,
       kExprCatchAll,
         kExprRethrow, 0x00,
       kExprEnd
     ])
     .exportFunc();
 
   const buffer = builder.toBuffer();
+
+  // The exception object's identity should be preserved throw 'rethrow's in
+  // Wasm code. Do tests with a tag defined in JS.
```

Comment:
I'm finding it hard to follow the first sentence

Reply:
I committed @dschuff's suggestion: https://github.com/WebAssembly/exception-handling/pull/256#discussion_r1110065053 Is this clearer?